engine/resource/src/mount/mount_android.cpp

```diff
@@ -122,11 +124,27 @@ namespace dmResource
         uint32_t data_length = 0;
         void* data_map = 0x0;
 
-        Result r = MapAsset(am, data_path, (void*&)data_asset, data_length, data_map);
-        if (r != RESULT_OK)
+        Result r;
+
+        // Map data asset (if bundled) or file (if in local storage)
+        bool mem_mapped_data = false;
+        if (strcmp(data_path, "game.arcd") == 0)
         {
-            dmLogError("Error when mapping data file '%s', result = %i", data_path, r);
-            return RESULT_IO_ERROR;
+            r = MapAsset(am, data_path, (void*&)data_asset, data_length, data_map);
+            if (r != RESULT_OK)
+            {
+                dmLogError("Error when mapping data file '%s', result = %i", data_path, r);
+                return RESULT_IO_ERROR;
+            }
+        } else
+        {
+            r = MapFile(data_path, data_map, data_length);
+            if (r != RESULT_OK)
+            {
+                dmLogError("Error mapping liveupdate data file, result = %i", r);
+                return RESULT_IO_ERROR;
+            }
+            mem_mapped_data = true;
```

Comment:
My use case was simple:
1. Download resource -> we create liveupdate.arcd/.arci
2. Reboot
3. Fails to MapAsset when using file on disc (liveupdate.arcd)